main.py

```diff
@@ -131,73 +145,147 @@ def calculate_interest(self, rate: float) -> None:
         print("Debit accounts do not earn interest.")
 
 
-class Customer:
-    def __init__(self, account_number: str, account: Account) -> None:
+class SavingsAccount:
+    def __init__(self, account_number: str, balance: float = 0) -> None:
         self.account_number = account_number
-        self.account = account
+        self.balance = balance
+        self.is_locked = False
+        self.transactions = []
+
+    def deposit(self, amount: float) -> None:
+        if not self.is_locked and amount > 0:
+            self.balance += amount
+            self.transactions.append(Transaction("Deposit", amount, datetime.datetime.now()))
+            print(f"Deposited ${amount}. New balance: ${self.balance}")
+        elif self.is_locked:
+            print("Account is locked. Cannot perform transactions.")
+        else:
+            print("Invalid deposit amount.")
```

Comment:
The self.is_locked condition could be extracted before this ifelse:
```
if self.is_locked: 
    print(...)
    return
```
That imo would improve readability